lang/csharp/src/apache/ipc/HttpTransceiver.cs

```diff
@@ -84,21 +102,26 @@ public static IList<MemoryStream> ReadBuffers(Stream inStream, byte[] intBuffer)
                 if (length == 0) //end of transmission
                     break;
 
-                byte[] buffer = new byte[length];
-                int offset = 0;
-                int count = length;
-                while (offset < length)
+                MemoryStream outStream = new MemoryStream(length);
+                byte[] buffer = new byte[64 * 1024];
+                int read;
+                int totalRead = 0;
+                while ((read = inStream.Read(buffer, 0, Math.Min(length - totalRead, buffer.Length))) > 0)
```

Comment:
This line sometimes stalls and causes the timeout failure in the HttpClientServerTest.cs that was mentioned in the [Jira ticket comments](https://issues.apache.org/jira/browse/AVRO-1910). I think that changing it to  
while ((length > totalRead)&&(read = inStream.Read(buffer, 0, Math.Min(length - totalRead, buffer.Length))) > 0) 
prevents the timeout on this line. But the test doesn't pass - it just times out later.